package/MDAnalysis/lib/_cutil.pyx

```diff
@@ -65,6 +78,213 @@ def unique_int_1d(np.ndarray[np.int64_t, ndim=1] values):
+       | !       ! |     |         ! | !
+       |-5-8   1-2-| ->  |       1-2-|-5-8
+       | !       ! |     |         ! | !
+       | 7       4 |     |         4 | 7
+       |           |     |           |
+       +-----------+     +-----------+
+
+
+    Parameters
+    ----------
+    atomgroup : AtomGroup
+        The :class:`MDAnalysis.core.groups.AtomGroup` to work with.
+        The positions of this are modified in place.  All these atoms
+        must belong in the same molecule or fragment.
+    reference_atom : :class:`~MDAnalysis.core.groups.Atom`
+        The atom around which all other atoms will be moved.
+        Defaults to atom 0 in the atomgroup.
+
+    Raises
+    ------
+    NoDataError
+        There are no bonds present.
+        (See :func:`~MDAnalysis.topology.core.guess_bonds`)
+
+    ValueError
+        The algorithm fails to work.  This is usually
+        caused by the atomgroup not being a single fragment.
+        (ie the molecule can't be traversed by following bonds)
+
+
+    Example
+    -------
+    Make fragments whole::
+
+        from MDAnalysis.lib.mdamath import make_whole
+
+        # This algorithm requires bonds, these can be guessed!
+        u = mda.Universe(......, guess_bonds=True)
+
+        # MDAnalysis can split molecules into their fragments
+        # based on bonding information.
+        # Note that this function will only handle a single fragment
+        # at a time, necessitating a loop.
+        for frag in u.fragments:
+          make_whole(frag)
+
+    Alternatively, to keep a single atom in place as the anchor::
+
+        # This will mean that atomgroup[10] will NOT get moved,
+        # and all other atoms will move (if necessary).
+        make_whole(atomgroup, reference_atom=atomgroup[10])
+
+
+    .. versionadded:: 0.11.0
+    """
+    cdef intset agset, refpoints, todo, done
+    cdef int i, nloops, ref, atom, other
+    cdef intmap bonding
+    cdef int[:, :] bonds
+    cdef float[:, :] oldpos, newpos
+    cdef bint ortho
+    cdef float[:] box
+    cdef float inverse_box[3]
+    cdef coordinate tri_box[3]
+    cdef double vec[3]
+
+    if reference_atom is None:
+        ref = atomgroup[0].index
+    else:
+        # Sanity check
+        if not reference_atom in atomgroup:
+            raise ValueError("Reference atom not in atomgroup")
+        ref = reference_atom.index
+
+    box = atomgroup.dimensions
+
+    for i in range(3):
+        if box[i] == 0.0:
+            raise ValueError("One or more dimensions was zero.  "
+                             "You can set dimensions using 'atomgroup.dimensions='")
+
+    ortho = True
+    for i in range(3):
+        if box[i + 3] != 90.0:
+            ortho = False
+
+    if ortho:
+        for i in range(3):
+            inverse_box[i] = 1.0 / box[i]
+    else:
+        from .mdamath import triclinic_vectors
+        # forgive me code Jesus for I have sinned
+        B = triclinic_vectors(box)
```

Comment:
does anyone know how we'd turn a 3,3 numpy array into the `coordinate[3]` type we need for the `pbc_triclinic` below?

Reply:
see [libmdaxdr](https://github.com/MDAnalysis/mdanalysis/blob/060e2c228f2b176421f1eab0db2513ba7a9cb1c3/package/MDAnalysis/lib/formats/libmdaxdr.pyx#L790)
`<coordinate*>&box[0, 0]`

Reply:
Oh to the typecasting on the B of course.

Reply:
The line takes the first pointer to the first element of `box` aka `box[0, 0]` and casts that to a pointer of type coordinate.